### ISSUE TITLE
[FEATURE] Allow separate schemes for LAPI and Appsec endpoints

### ISSUE THREAD

**Issue by thespad** [maxlerebourg/crowdsec-bouncer-traefik-plugin#131]
**Is your feature request related to a problem? Please describe.** 🐛
Currently (or rather in the RC) the appsec and LAPI endpoints both use `crowdsecLapiScheme` https://github.com/maxlerebourg/crowdsec-bouncer-traefik-plugin/pull/123/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R79 but there are scenarios where this doesn't work - such as pointing at a central LAPI service served over https and a local appsec service over an internal docker network served over http.

**Describe the solution you'd like** ✨
Allow separate schemes for LAPI and Appsec endpoints

**Comment by mathieuHa** [comment]
Hi,

Thanks for using the plugin.  

When we added the support for appsec, we thought about "scheme" to connect to the LAPI and Appsec.  

Because we already have a lot of variables we decided to not add support by default for different supported scheme.  
We wanted to see if it was really needed before implementing it.  

I believe it needs at least 2 variables, scheme + insecure and 2 objects in the bouncer httpClient containing a tlsconfig.
It is added to the todolist, we will update you soon

**Comment by thespad** [comment]
Yeah, I understand the desire not to add a bunch more configuration options but unfortunately the way Crowdsec have implemented their appsec component it's perfectly legitimate to have your bouncer talking to a Central LAPI endpoint for normal decisions but a local crowdsec instance for appsec - and in a docker setup that means that configuring https for that appsec instance is a lot of work for little gain because you can push it all over a private docker network.

In the meantime I'm sending all my appsec traffic back to the central LAPI to keep everything secure, and the latency isn't a huge issue in my setup, but it would be nice to be able to split them up.

**Comment by mathieuHa** [comment]
Hi @thespad,

This has been added to the todo of the project. Will work on it when we got the time

**Comment by smeevil** [comment]
Looking forward to this change as well. We use the setup in K8S. The bouncer plugin connects to appsec directly, which runs in http, whereas LAPI runs in https. @thespad how did you setup it so that appsec is routed through LAPI?

**Comment by thespad** [comment]
I'm just doing

```yaml
          crowdsecAppsecEnabled: true
          crowdsecAppsecHost: crowdsec-appsec.example.com
          crowdsecLapiHost: crowdsec.example.com
          crowdsecLapiScheme: https
```

Both of those domains resolve to the crowdsec LAPI host but send the traffic to different ports as required.

**Comment by benjvfr** [comment]
Hello, I encounter exactly this issue using a distributed CrowdSec setup.

Is there any ETA on this feature ?

Thanks.

**Comment by andreasbrett** [comment]
> the way Crowdsec have implemented their appsec component it's perfectly legitimate to have your bouncer talking to a Central LAPI endpoint for normal decisions but a local crowdsec instance for appsec

Just wanted to mention that you lose the ability of appsec sending ban decisions to your LAPI for out-of-band appsec rules. The appsec instance would send it to its own LAPI. Thus the ban decision (because of multiple out-of-band violations) will not be enacted on the LAPI that the traefik middleware fetches decisions from.
